packages/components/src/internal/components/user/ProfilePage.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2019 LabKey Corporation. All rights reserved. No portion of this work may be reproduced in
+ * any form or by any electronic or mechanical means without written permission from LabKey Corporation.
+ */
+import React, { FC, useState } from 'react';
+import { Button } from 'react-bootstrap';
+
+import { createNotification, withTimeout } from '../notifications/actions';
+import { queryGridInvalidate } from '../../actions';
+import { SCHEMAS } from '../../schemas';
+import { isLoginAutoRedirectEnabled } from '../administration/utils';
+import { InsufficientPermissionsPage } from '../permissions/InsufficientPermissionsPage';
+
+import { Page } from '../base/Page';
+
+import { Section } from '../base/Section';
+
+import { Notification } from '../notifications/Notification';
+
+import { useServerContext } from '../base/ServerContext';
+
+import { App } from '../../../index';
+
+import { UserDetailHeader } from './UserDetailHeader';
+import { getUserRoleDisplay } from './actions';
+
+import { UserProfile } from './UserProfile';
+import { ChangePasswordModal } from './ChangePasswordModal';
+
+import { useUserProperties } from './UserProvider';
+
+interface Props {
+    updateUserDisplayName: (displayName: string) => any;
+    setReloadRequired: () => any;
+    goBack: (n?: number) => any;
+}
+
+const TITLE = 'User Profile';
+
+export const ProfilePage: FC<Props> = props => {
+    const [showChangePassword, setShowChangePassword] = useState<boolean>(false);
+    const { user } = useServerContext();
+    const userProperties = useUserProperties(user);
+
+    if (!user.isSignedIn) {
+        return <InsufficientPermissionsPage title={TITLE} />;
+    }
+
+    const navigate = (result: {}, shouldReload: boolean): void => {
+        const { goBack, setReloadRequired, updateUserDisplayName } = props;
+        const successMsg = 'Successfully updated your user profile.';
+
+        if (shouldReload) {
+            createNotification(successMsg);
```

Comment:
nit: I realize this is likely never true but the logic of this method could result in `createNotification(successMsg)` being called twice if `shouldReload` is `true` and `result.success` is `true` as well. Just something I noticed.